### PR TITLE
Support SparseB for fp16/bf16/int8

### PR DESCRIPTION
**Pull request by vin-huang**

### CODE REVIEW

**Review by hcman2 — APPROVED** [review]
LGTM

File: tensilelite/Tensile/Common.py [hunk 1]
```diff
@@ -1121,7 +1121,7 @@ def getArchitectureName(gfxName):
     "HighPrecisionAccumulate":  False,            # f32 += f16*f16
     "SilentHighPrecisionAccumulate": False,       # Keep kernel names the same for HPA mode.  Useful for testing.
 
-    "SparseA":                  False,            # 4:2 Structured Sparse A Matrix
+    "SparseA":                  0,                # 4:2 Structured Sparse A Matrix, 0=Non Sparse, 1=Sparse Matrix A, 2=Sparse Matrix B
```
Comment (jichangjichang) [inline tensilelite/Tensile/Common.py:1124]: Don't use SparseA as KernelProblemParameter since it is used for both A and B.
Revise it as "Sparse" at least

Comment (vin-huang) [reply tensilelite/Tensile/Common.py]: done, please review again

File: tensilelite/Tensile/Common.py [hunk 2]
```diff
@@ -1121,7 +1121,7 @@ def getArchitectureName(gfxName):
     "HighPrecisionAccumulate":  False,            # f32 += f16*f16
     "SilentHighPrecisionAccumulate": False,       # Keep kernel names the same for HPA mode.  Useful for testing.
 
-    "SparseA":                  False,            # 4:2 Structured Sparse A Matrix
+    "SparseA":                  0,            # 4:2 Structured Sparse A Matrix
```
[the remark this answers is not in the document]
Comment (vin-huang) [reply tensilelite/Tensile/Common.py]: done, pleaes review again

File: tensilelite/Tensile/Common.py (hunk 1 above)
Comment (vin-huang) [reply tensilelite/Tensile/Common.py]: done, please review again

File: tensilelite/Tensile/Common.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (vin-huang) [reply tensilelite/Tensile/Common.py]: done, pleaes review again

File: tensilelite/Tensile/Source/lib/source/ContractionSolution.cpp [hunk 3]
```diff
@@ -605,14 +605,19 @@ namespace Tensile
 
         for(size_t i = startStrideAB; i < a.dimensions(); i++)
         {
-            auto stride_a = problemType.sparseA ? ca.strides()[i] : a.strides()[i];
+            auto stride_a
+                = problemType.sparse && problemType.sparse != 2 ? ca.strides()[i] : a.strides()[i];
```
Comment (jichangjichang) [inline tensilelite/Tensile/Source/lib/source/ContractionSolution.cpp:609]: condition checking should direct using "problemType.sparse == 1"
ca sould rename as compressMatrix or something else but better not "ca"

**Review by KKyang — APPROVED** [review]

**Pull request merged**